presto-hive/src/main/java/io/prestosql/plugin/hive/coercions/DecimalCoercers.java

```diff
@@ -321,4 +325,25 @@ protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int pos
                     realToLongDecimal(fromType.getLong(block, position), toType.getPrecision(), toType.getScale()));
         }
     }
+
+    public static Function<Block, Block> createDecimalToVarcharCoercer(DecimalType fromType, VarcharType toType)
+    {
+        return new DecimalToVarcharCoercer(fromType, toType);
+    }
+
+    private static class DecimalToVarcharCoercer
+            extends TypeCoercer<DecimalType, VarcharType>
+    {
+        public DecimalToVarcharCoercer(DecimalType fromType, VarcharType toType)
+        {
+            super(fromType, toType);
+        }
+
+        @Override
+        protected void applyCoercedValue(BlockBuilder blockBuilder, Block block, int position)
+        {
+            toType.writeSlice(blockBuilder,
+                    utf8Slice(new BigDecimal(BigInteger.valueOf(fromType.getLong(block, position)), fromType.getScale()).toString()));
```

Comment:
Any better way of coercing without going through so many allocations?
For other types I was able to find some static helper methods.

Also, do we need to verify if the string representation of the DECIMAL can fit in the bounded length defined for the VarcharType? Or will such cases automatically throw proper exceptions higher in the stack.

Reply:
> should we verify if the string representation of the DECIMAL can fit in the bounded length defined for the VarcharType? 

would this also be a problem with other varchar coercions? (eg. existing  INTEGER to varchar coercer for varchar(1) ) We could verify the exception propagation in that case and handle all of them in a separate commit or a follow-up PR.

Reply:
I'm not sure about the semantics of this conversion, and whether we need to do it separately for short and long decimals. (There're some helper methods in `Decimals` class for this conversion)

Reply:
Thanks for the `Decimals` class. Other coercions don't explicitly handle the Varchar bounds too.